### PR TITLE
Add basic vue component unit tests

### PR DESCRIPTION
**Pull request by davidrapson**
This PR puts tooling in place to be able to unit test our vue components. Definitely not comprehensive yet but adds an example test for:

- Basic interactions by toggling state
- Snapshot test for a functional component
- Snapshot test for a full component

I'll need to rework this to account for translations, but review still welcome now.

### CODE REVIEW

**Comment by davidrapson** [comment]
Spotted as part of this that whilst we have https://github.com/vuejs/eslint-plugin-vue installed it wasn't actually running against `.vue` files 🙈 . Fixed that and auto-fixed a few warnings, as well as disabling any rules that are overkill or are handled by prettier. Leaves a few warnings about prop-types which sound sensible but I'll come back and fix.

**Comment by davidrapson** [comment]
@mattandrews Mind giving this another look 👀

File: assets/.eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 module.exports = {
-    extends: ['eslint:recommended', 'plugin:vue/recommended', 'plugin:compat/recommended'],
+    extends: ['eslint:recommended', 'plugin:vue/strongly-recommended', 'plugin:compat/recommended'],
```
Comment (mattandrews) [inline assets/.eslintrc.js]: 💪

Comment (davidrapson) [reply assets/.eslintrc.js]: Weirdly "strongly-recommended" is weaker than "recommended" 🤷‍♂️

**Pull request merged**